tutorials/adding_new_layers.md

```diff
@@ -54,33 +57,200 @@
 - Open [filterbar.js](https://github.com/Cloud-CV/Fabrik/blob/master/ide/static/js/filterbar.js) in a text editor, add ```"your_layer_id"``` to 1(or more) of 3 framework filter array ```var KerasLayers = [...]```, ```var TensorFlowLayers = [...]``` or ```var CaffeLayers = [...]```. This should be like this ```var KerasLayers = ["RNN_Button", "GRU_Button", "your_layer_id"]```. This arrays are placed inside ```changeEvent() {}``` function.
 
-### Adding layer handling to the backend
+
+## Adding layer handling to the backend
+
+### 1. Caffe
+
+#### Import
 
 - Open [import_prototxt.py](https://github.com/Cloud-CV/Fabrik/blob/master/caffe_app/views/import_prototxt.py) file in a text editor.
 
-<img src="https://raw.githubusercontent.com/Cloud-CV/Fabrik/master/tutorials/layerImportPrototxt1.png" />
+    <img src="https://raw.githubusercontent.com/Cloud-CV/Fabrik/master/tutorials/layerImportPrototxt1.png" />
 
 - Add a function for the new layer below the category of this layer.
 - Load the parameters, do the calculations for your layer in pyhton and return the value of ```params``` (parameters).
 - Move down in the file.
 
-<img src="https://raw.githubusercontent.com/Cloud-CV/Fabrik/master/tutorials/layerImportPrototxt2.png" />
+    <img src="https://raw.githubusercontent.com/Cloud-CV/Fabrik/master/tutorials/layerImportPrototxt2.png" />
 
 - Add your defined layer in the ```layer_dict``` array, as shown above.
 
+#### Export
+
 - Now, open [jsonToPrototxt.py](https://github.com/Cloud-CV/Fabrik/blob/master/ide/utils/jsonToPrototxt.py) in a text editor.
 
-<img src="https://raw.githubusercontent.com/Cloud-CV/Fabrik/master/tutorials/layerJSONtoPrototxt1.png" />
+    <img src="https://raw.githubusercontent.com/Cloud-CV/Fabrik/master/tutorials/layerJSONtoPrototxt1.png" />
 
 - Add an export function for training and testing of the new layer.
 - There you need to load parameters, then train & test values and at last return the trained and tested data.
 - Move down in this file as well.
 
-<img src="https://raw.githubusercontent.com/Cloud-CV/Fabrik/master/tutorials/layerJSONtoPrototxt2.png" />
+    <img src="https://raw.githubusercontent.com/Cloud-CV/Fabrik/master/tutorials/layerJSONtoPrototxt2.png" />
 
 - Add the export function in the ```layer_map``` array.
 
-### Testing and pushing the new layer.
+
+### Keras
+
+#### Importing a layer
+
+- Open [keras_app/views/layers_import.py](https://github.com/Cloud-CV/Fabrik/blob/master/keras_app/views/layers_import.py).
+    - Add a function to process the new layer.
+        - The function should take a Keras `Layer` object, called `layer`.
+        - You can get the layer parameters from `layer`, and create a json layer with the function  [jsonLayer](https://github.com/Cloud-CV/Fabrik/blob/master/keras_app/views/layers_import.py#L529). 
+        - Return the json layer you just created.
+
+        ```
```

Comment:
Please maintain consistency in style for whole document. As there are images in above explanation for code can you either add images here or change the other images to code.

Reply:
I've changed all the images to texts whenever possible